File: 01. JAVA/01. SERVIDORES/transaccion/src/main/java/edu/ec/monster/transaccion/model/Transaction.java
package edu.ec.monster.transaccion.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "transactions")
public class Transaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "transaction_type", nullable = false)
    private String transactionType;  // 'retiro', 'deposito', 'transferencia'

    @Column(name = "amount", nullable = false)
    private BigDecimal amount;

    @Column(name = "source_account_id")
    private Long sourceAccountId;

    @Column(name = "destination_account_id")
    private Long destinationAccountId;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(name = "transaction_date")
    private LocalDateTime transactionDate = LocalDateTime.now();

    @Column(name = "status")
    private String status;

}

File: 02. DOTNET/02. CLI WEB/src/models/Transaction.js
// src/models/Transaction.js
export default class Transaction {
    constructor(id, transactionType, amount, sourceAccountId, destinationAccountId, userId, transactionDate, status) {
      this.id = id;
      this.transactionType = transactionType;
      this.amount = amount;
      this.sourceAccountId = sourceAccountId;
      this.destinationAccountId = destinationAccountId;
      this.userId = userId;
      this.transactionDate = transactionDate;
      this.status = status;
    }
  
    static fromApiResponse(response) {
      return response.map(item => new Transaction(
        item.id,
        item.transactionType,
        item.amount,
        item.sourceAccountId,
        item.destinationAccountId,
        item.userId,
        item.transactionDate,
        item.status
      ));
    }
  }
  